runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftOverlay.java

```diff
@@ -77,36 +86,83 @@ public Dimension render(Graphics2D graphics)
 
 		for (WidgetItem item : widgetItems)
 		{
+			// if stays 0, then pouch is broken
+			int maxEssence = 0;
 			Varbits varbits;
 
 			switch (item.getId())
 			{
 				case ItemID.SMALL_POUCH:
 					varbits = Varbits.POUCH_SMALL;
+					maxEssence = 3;
 					break;
 				case ItemID.MEDIUM_POUCH:
+					varbits = Varbits.POUCH_MEDIUM;
+					maxEssence = 6;
+					break;
 				case MEDIUM_POUCH_DAMAGED:
 					varbits = Varbits.POUCH_MEDIUM;
 					break;
 				case ItemID.LARGE_POUCH:
+					varbits = Varbits.POUCH_GIANT;
+					maxEssence = 9;
+					break;
 				case LARGE_POUCH_DAMAGED:
 					varbits = Varbits.POUCH_LARGE;
 					break;
 				case ItemID.GIANT_POUCH:
+					varbits = Varbits.POUCH_GIANT;
+					maxEssence = 12;
+					break;
 				case GIANT_POUCH_DAMAGED:
 					varbits = Varbits.POUCH_GIANT;
 					break;
 				default:
 					continue;
 			}
 
-			final Rectangle bounds = item.getCanvasBounds();
-			final TextComponent textComponent = new TextComponent();
-			textComponent.setPosition(new Point(bounds.x, bounds.y + 16));
-			textComponent.setText(String.valueOf(client.getVar(varbits)));
-			textComponent.render(graphics);
+			final int essenceInPouch = client.getVar(varbits);
+
+			if (config.overlayStyle() != OverlayStyle.NONE)
+			{
+				if (maxEssence == 0)
+				{
+					renderOverlay(graphics, item, Color.RED, TRANSPARENT_RED);
+				}
+				else if (essenceInPouch == maxEssence)
+				{
+					renderOverlay(graphics, item, Color.GREEN, TRANSPARENT_GREEN);
+				}
+				else
+				{
+					renderOverlay(graphics, item, Color.YELLOW, TRANSPARENT_YELLOW);
+				}
+			}
+
+			if (config.showPouchCount())
+			{
+				final Rectangle bounds = item.getCanvasBounds();
+				final TextComponent textComponent = new TextComponent();
+				textComponent.setPosition(new Point(bounds.x, bounds.y + 16));
+				textComponent.setText(String.valueOf(essenceInPouch));
+				textComponent.render(graphics);
+			}
 		}
 
 		return null;
 	}
+
+	private void renderOverlay(final Graphics2D graphics, final WidgetItem item, final Color outlineColor, final Color backgroundColor)
+	{
+		if (config.overlayStyle() == OverlayStyle.OUTLINE)
+		{
+			final BufferedImage outline = itemManager.getItemOutline(item.getId(), item.getQuantity(), outlineColor);
+			graphics.drawImage(outline, item.getCanvasLocation().getX() + 1, item.getCanvasLocation().getY() + 1, null);
```

Comment:
Are you trying to do a drop shadow effect? How do you explain this plus one offset?

Reply:
Yep, as @deathbeam said in Discord, this is just how the outline effect works to be aligned properly. It's the same for the inventory tag plugin.

Reply:
Would be nice for a comment explaining the plus one.
The plus sixteen above too.